### PR TITLE
chore: add swagger authentication spec [DET-4272]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

this lets generated code through swagger-code-gen know how to apply authentication

## Test Plan

- tested using the example here https://github.com/determined-ai/api-bindings

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by hamidzr** [comment]
@hkang1 webui wasn't running into this authentication issue because the browser carries over the cookie on its own and we still accept that as an authentication method.

File: proto/src/determined/api/v1/api.proto [hunk 1]
```diff
@@ -21,33 +22,59 @@ import "determined/api/v1/shell.proto";
...
 };
 
 // Determined is the official v1 of the Determined API.
 service Determined {
     // Login the user.
     rpc Login(LoginRequest) returns (LoginResponse) {
         option (google.api.http) = {post: "/api/v1/auth/login" body: "*"};
-        option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {tags: "Authentication"};
+        option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {tags: "Authentication",
+            security: {};};
```
Comment (hkang1) [inline proto/src/determined/api/v1/api.proto:77]: **question:** the semi-colon `;` after `security: {}` is probably a typo/unnecessary?

Comment (hkang1) [reply proto/src/determined/api/v1/api.proto:77]: either that or the `,` after `tags` needs to be a semi-colon?

Comment (hamidzr) [reply proto/src/determined/api/v1/api.proto:77]: it's pretty loose they all should probably be semicolons as these are not really JSON blobs and grpc_gateway examples use semicolon. I have a ticket for adding a formatter that should autoformat all of these.

Comment (hamidzr) [reply proto/src/determined/api/v1/api.proto:77]: I'll go ahead and make sure the parts I change stay consistent with semicolon vs `comma`

File: webui/react/src/services/api.ts [hunk 2]
```diff
@@ -20,11 +20,11 @@ import { decodeExperimentList, encodeExperimentState } from './decoder';
 
 export { isAuthFailure, isLoginFailure, isNotFound } from './utils';
 
-const address = serverAddress();
+const ApiConfig : Api.Configuration = { basePath: serverAddress() };
```
Comment (hkang1) [inline webui/react/src/services/api.ts:23]: **question:** I'm not sure where the changes are on the swagger end that causes this change. I like this much better than having to pass in an `undefined` argument, but just not sure what resulted in this usage change.

**Review by hkang1 — COMMENTED** [review]
Couple of questions I wasn't sure about.

File: proto/src/determined/api/v1/api.proto [hunk 3]
```diff
@@ -21,33 +22,59 @@ import "determined/api/v1/shell.proto";
...
+    license : {
+      name : "Apache 2.0",
+      url : "http://www.apache.org/licenses/LICENSE-2.0.html"
+    }
+  }
+  schemes : [ HTTP, HTTPS ],
+  external_docs : {
+    description : "Determined AI Documentation",
+    url : "/docs",
+  },
```
Comment (hkang1) [inline proto/src/determined/api/v1/api.proto:54]: guessing commas are optional? just seeing inconsistencies in the protobuf. For example, `external_docs` has a comma at the end of the dictionary, but not for `license`, `contact` nor `info` (and other places too)

File: webui/react/src/services/api.ts (hunk 2 above)
Comment (hamidzr) [reply webui/react/src/services/api.ts:23]: nothing this is a no-op. I had a TODO here that I removed. but basically once, and if, we start to use the intended way of authentication with the generated code we'd need this configuration file.

File: proto/src/determined/api/v1/api.proto (hunk 3 above)
Comment (hamidzr) [reply proto/src/determined/api/v1/api.proto:54]: yeah optional I couldn't find a reference to a formatting guide that mentions that but this is how grpc_gateway people do it. https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/a_bit_of_everything.proto

**Review by hkang1 — APPROVED** [review]
LGTM! Thanks for answering my questions.

File: proto/src/determined/api/v1/api.proto (hunk 1 above)
Comment (hkang1) [reply proto/src/determined/api/v1/api.proto]: and I thought JSON was bad... :D

**Pull request merged**